src/wp-admin/includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,634 @@
+					esc_html__( 'Plugin %1$s is a dependency for the following plugins: %2$s.' ),
+					esc_html( $plugin_data['Name'] ),
+					esc_html( implode( ', ', $parents_names ) )
+				) : sprintf(
+					/* translators: %1$s: plugin name. %2$s: Parent plugin name. */
+					esc_html__( 'Plugin %1$s is a dependency for the "%2$s" plugin.' ),
+					esc_html( $plugin_data['Name'] ),
+					esc_html( $parents_names[0] )
+				);
+
+			$this->inline_plugin_row_notice( $notice_contents, 'info', $plugin_file );
+		}
+
+		// Add extra info to parents.
+		if ( $has_dependencies ) {
+			$style = is_rtl() ? 'border-top:none;border-left:none' : 'border-top:none;border-right:none';
+			if ( $pending_activation ) {
+				if ( $in_circular_dependency ) {
+					$this->inline_plugin_row_notice(
+						sprintf(
+							/* translators: %s: plugin name. */
+							esc_html__( 'Warning: Circular dependencies detected. Plugin "%s" has unmet dependencies.' ),
+							esc_html( $plugin_data['Name'] )
+						),
+						'warning',
+						$plugin_file
+					);
+				} else {
+					$this->inline_plugin_row_notice(
+						sprintf(
+							/* translators: %s: plugin name. */
+							esc_html__( 'Plugin "%s" has unmet dependencies. Once all required plugins are installed the plugin will be automatically activated. Alternatively you can cancel the activation of this plugin by clicking on the "cancel activation request" link above.' ),
+							esc_html( $plugin_data['Name'] )
+						),
+						'warning',
+						$plugin_file
+					);
+				}
+			} elseif ( ! $is_plugin_active ) {
+				$dependencies_human_readable = array();
+				foreach ( $dependencies as $dependency ) {
+					$plugin_file = $this->get_plugin_file_from_slug( $dependency['slug'] );
+					if ( $plugin_file ) {
+						$dependencies_human_readable[] = get_plugin_data( WP_PLUGIN_DIR . '/' . $plugin_file )['Name'];
+					} else {
+						$dependencies_human_readable[] = $dependency['slug'];
+					}
+				}
+				$this->inline_plugin_row_notice(
+					sprintf(
+						/* translators: %1$s: plugin name. %2$s: plugin requirements, comma-separated. */
+						esc_html__( 'Plugin "%1$s" depends on the following plugin(s): %2$s' ),
+						esc_html( $plugin_data['Name'] ),
+						esc_html( implode( ', ', $dependencies_human_readable ) )
+					),
+					'info',
+					$plugin_file
+				);
+			}
+		}
+	}
+
+	/**
+	 * Generate the contents of an inline plugin row notice.
+	 *
+	 * @since 5.9.0
+	 * @access protected
+	 */
+	protected function inline_plugin_row_notice( $contents = '', $notice_type = 'info', $plugin_file = '' ) {
+		$is_plugin_active = is_plugin_active( $plugin_file );
+		$tr_class         = $is_plugin_active ? 'plugin-dependencies-tr active' : 'plugin-dependencies-tr';
+		$td_class         = 'plugin-dependencies colspanchange';
+		$colspan          = (int) _get_list_table( 'WP_Plugins_List_Table', array( 'screen' => get_current_screen() ) )->get_column_count();
```

Comment:
This part is confusing to me. The `_get_list_table` will return either list table object on success, false if the class does not exist. By casting it to integer, you're trying to get either 1 or 0?

But in either case, casting an object to an integer should throw a notice in PHP. Am I missing something?

Reply:
it's not casting the list table object, it's casting `$list_table_object->get_column_count()`.

Maybe if it were rewritten as `(int) ( _get_list_table( 'WP_Plugins_List_Table', array( 'screen' => get_current_screen() ) ) )->get_column_count();` it would be more clear?

Reply:
now that I think about it, on all core screens with list tables, the list table object is stored in the `$wp_list_table` global.

this line should be be `$colspan = $wp_list_table->get_column_count()` (with `global $wp_list_table;` added at the start of the method, and the DocBlock updated appropriately).

Reply:
Oh, I totally missed the `->` part!

From the documentation the result of `get_column_count` is subtraction of two `count` functions. And that should always be an integer? 

> Returns the number of elements in value. When the parameter is neither an array nor an object with implemented Countable interface, 1 will be returned. There is one exception, if value is null, 0 will be returned.

I tried putting `null`, but besides warnings and errors in PHP 7.2+, the result is always an integer.

So it seems like the casting is not necessary 🤷🏼‍♂️

Reply:
> now that I think about it, on all core screens with list tables, the list table object is stored in the $wp_list_table global.

Yeah, tried that... Unfortunately `$wp_list_table` here is `null` because if I'm not mistaken this runs _before_ the list. So I wrote it like this - taken from https://github.com/WordPress/wordpress-develop/blob/c6a9a598bb956d688fcee0b56277f0be4d9e5dec/src/wp-admin/includes/update.php#L458-L463

> So it seems like the casting is not necessary 🤷🏼‍♂️

True, it's not necessary. I only added that as a better-safe-than-sorry measure, but we can definitely remove it if it doesn't serve a purpose!